docs/preview/02-Features/02-message-handling/03-event-hubs.md

```diff
@@ -0,0 +1,131 @@
+- **Message pump**: implementation that interacts and receives the Azure EventHubs event message. The pump can be configured for different scenarios, see [this section](#pump-configuration) for more information.
+
+![Message handling schema](/media/worker-eventhubs-message-handling.png)
+
+## Installation
+This features requires to install our NuGet package:
+
+```shell
+PM > Install-Package Arcus.Messaging.Pumps.EventHubs
+```
+
+## Message handler example
+Here is an example of a message handler that expects messages of type `Order`:
+
+```csharp
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Pumps.EventHubs;
+using Microsoft.Extensions.Logging;
+
+public class OrdersMessageHandler : IAzureEventHubsMessageHandler<Order>
+{
+    private readonly ILogger _logger;
+
+    public OrdersMessageHandler(ILogger<OrdersMessageHandler> logger)
+    {
+        _logger = logger;
+    }
+
+    public async Task ProcessMessageAsync(
+        Order orderMessage, 
+        AzureEventHubsMessageContext messageContext, 
+        MessageCorrelationInfo correlationInfo, 
+        CancellationToken cancellationToken)
+    {
+        _logger.LogInformation("Processing order {OrderId} for {OrderAmount} units of {OrderArticle} bought by {CustomerFirstName} {CustomerLastName}", orderMessage.Id, orderMessage.Amount, orderMessage.ArticleNumber, orderMessage.Customer.FirstName, orderMessage.Customer.LastName);
+
+        // Process the message.
+
+        _logger.LogInformation("Order {OrderId} processed", orderMessage.Id);
+    }
+}
+```
+
+## Message handler registration
+Once the message handler is created, you can very easily register it:
+
+```csharp
+using Microsoft.Extensions.DependencyInjection;
+
+public class Program
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        // Arcus secret store will be used to lookup the connection strings, 
+        // for more information about the Arcus secret store see: https://security.arcus-azure.net/features/secret-store
+        services.AddSecretStore(stores => ...);
+
+        // Add Azure EventHubs message pump and use OrdersMessageHandler to process the messages.
+        services.AddEventHubsMessagePump("<my-eventhubs-name>", "Arcus_EventHubs_ConnectionString", "<my-eventhubs-blob-storage-container-name>", "Arcus_EventHubs_Blob_ConnectionString")
+                .WithEventHubsMessageHandler<OrdersMessageHandler, Order>();
+
+        // Note, that only a single call to the `.WithEventHubsMessageHandler` has to be made when the handler should be used across message pumps.
+    }
+}
+```
+
+The Azure EventHubs uses the `EventProcessorClient` internally. To learn more about this way of consuming messages from Azure EventHubs, see [Microsoft's official documentation](https://docs.microsoft.com/en-us/dotnet/api/overview/azure/messaging.eventhubs.processor-readme).
```

Comment:
I was aware of the EventProcessorHost (used it a few years ago).  EventProcessorClient is something new I think ?
/just fmi.
Seems to be a new approach ? https://gist.github.com/ShivangiReja/18d8e52a538c0890567fef4177306cb3

/ sorry for the noise :D

Reply:
Yes, with the move to the new Azure SDK, there are a lot of new namespaces and types. Seemed that it was good that we already made use of the new SDK as, I imagine, the older one (with the `Microsoft.Bla.Bla` namespace instead of the `Azure.Bla.Bla` namespace) will be deprecated.
We already see this in our security repo where we have to move to the new Azure SDK.

Reply:
> / sorry for the noise :D

No problem 😉 , best that you are still up-to-date with what's happening and what we're using.